Dockerfile

```diff
@@ -0,0 +1,51 @@
+#Dockerfile
+
+# base image
+FROM python:3.11.2 as base
+
+ENV PYTHONUNBUFFERED=1 \
+    # prevents creating .pyc files
+    PYTHONDONTWRITEBYTECODE=1 \
+    PIP_NO_CACHE_DIR=off \
+    PIP_DISABLE_PIP_VERSION_CHECK=on \
+    PIP_DEFAULT_TIMEOUT=100 \
+    POETRY_HOME="/opt/poetry" \
+    POETRY_VIRTUALENVS_IN_PROJECT=true \
+    POETRY_NO_INTERACTION=1 \
+    PYSETUP_PATH="/opt/pysetup" \
+    VENV_PATH="/opt/pysetup/.venv"
+
+ENV PATH="$POETRY_HOME/bin:$VENV_PATH/bin:$PATH"
+
+# Build dependencies & install curl 
+FROM base as builder
+RUN apt-get update \
+    && apt-get install --no-install-recommends -y \
+        curl \
+        build-essential
+
+# Install poetry
+ENV POETRY_VERSION=1.4.1
+RUN curl -sSL https://install.python-poetry.org | python -
+
+WORKDIR $PYSETUP_PATH
+COPY ./poetry.lock ./pyproject.toml ./
+RUN poetry install --no-dev
+
+# production 
+FROM base as production
+ENV PATH="/opt/pysetup/.venv/bin:$PATH"
+WORKDIR /realtor
+
+COPY --from=builder $VENV_PATH $VENV_PATH
+COPY ./gunicornConf.py /gunicornConf.py
+
+COPY ./docker-entrypoint.sh /docker-entrypoint.sh
+RUN chmod +x /docker-entrypoint.sh
+
+COPY ./realtor /realtor
```

Comment:
이 부분을 비롯해서 repo에 있는 파일에 대한 불필요한 COPY가 많이 보입니다. docker-compose.yaml의 volume을 적절히 설정하면 피할 수 있어 보입니다.

참고로 저희 회사에서 장고로 되어 있는 컴포넌트 중 이렇게 볼륨을 정의한 예시도 보입니다.

```yaml
volumes:
  - .:/app
```

이 리포지토리도 마찬가지로 Dockerfile과 docker-compose.yaml이 root directory에 있는 등 현재 프로젝트와 유사한 구조를 가지고 있습니다.

Reply:
볼륨 - 컨테이너의 생명주기에 상관없이 데이터를 영속적으로 저장. 볼륨/바인드를 통하지 않은 데이터는 컨테이너가 종료될 때 사라짐. 
볼륨 lets a container run from your image have access to some path on whatever local machine the container is being run on. You cannot use files from your VOLUME directory in your Dockerfile. Anything in your volume directory will not be accessible at build-time but will be accessible at run-time.


찾아보니 local development에서는 volume deployment copy를 추천한다고 나와있네요. because "as the volume is actually creating a mounting point to the host. So if you are doing development where you need to change the code, and you use COPY you have to re-build the image everytime you want to have your changes to be in the container, whereas with volume you do not have to do rebuild your image the changes will be transfered to the container".